### ISSUE TITLE
Staging Review finding: "On this page" links move viewport but don't move keyboard focus

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#103578]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Education Data Migration
**Product:** Yellow Ribbon Tool
**Feature:** Yellow Ribbon Tool

## Findings details
**VA.gov Experience Standard - issue:** [User experiences predictable interactions with components and patterns.](https://design.va.gov/about/experience-standards/#user-can-identify-an-element-and-its-state)
**VA.gov Experience Standard - category:** [Usability](https://design.va.gov/about/experience-standards/#category-usability)
**Launch-blocking:** Yes
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

When you activate the "Yellow Ribbon Program information" link (or any of the links in the "On this page" section), the viewport shifts to the correct section of the screen but keyboard focus remains on that link. This has a couple of significant accessibility consequences:

For sighted keyboard-only users, tabbing to "Yellow Ribbon Program information" and pressing [Enter] visually gets you to the right section of content. But pressing [tab] to interact with anything in the Yellow Ribbon Program section immediately jumps you back up the page to the other "On this page" links. That is potentially disorienting and definitely frustrating for keyboard-only users, who now have to press tab about 30 more times to actually get keyboard focus to the right place.

For screen reader users who can't see any of the visual cues (also typically keyboard-only users), tabbing to "Yellow Ribbon Program information" and pressing [Enter] results in... nothing --- or, rather, nothing announced by the screen reader. Since focus hasn't changed and nothing has prompted the screen reader to say anything, it's just silent. Silence can be incredibly disorienting and forces users to guess about what's going on and what they should do next. In general it's bad form to force users to guess about how interactions are supposed to work.

<details><summary>Link, screenshot or steps to recreate</summary>

https://github.com/user-attachments/assets/83a20586-79e8-456c-bcc0-eadcde3f960f

Steps to reproduce:
1. Visit the [NYU details page](https://staging.va.gov/education/gi-bill-comparison-tool/institution/31800132)
2. Press [tab] until focus is on "Yellow Ribbon Program information" in the "On this page" section
3. Press [enter]
4. Note that you've visually jumped down the page to the Yellow Ribbon Program section
5. Press [tab] to move to the next logical link
6. Note that you've jumped back up to the "On this page" section

</details>

## Recommended action

Links to `#anchor` sections on the page actually need to go to that anchor. This is the out-of-the-box functionality for a link with `href="#anchor"`, so I'm guessing there's some script that's interfering with default browser behavior.

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** N/A
- **Modality:** Keyboard
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

_How did this touchpoint go? [Give feedback on the Collaboration Cycle](https://dj540s05.optimalworkshop.com/questions/f54355271969f160ef41182ba9ceb61c) at any time._
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by cassidy-beach** [comment]
Addressed in EDM-798

![Image](https://github.com/user-attachments/assets/39270f42-9d55-4372-b8e2-4fd8b02153cc)
